cloud-exec.md

```diff
@@ -0,0 +1,153 @@
+# Cloud execution
+
+- Enhancement Proposal PR: (leave this empty)
+- Contributors: dmpetrov
+
+# Summary
+
+Users need to run ML experiments in cloud instances or remote machines.
+DVC should support this out of the box to improve user experience and remove
+  the overhead of instrumenting these scenarios.
+
+# Motivation
+
+Today users need to spend time on instrumenting remote execution when there is
+  a lack of local resources for ML trainig (like GPU or RAM) or when it is a
+  company policy to train on dedicated hardware or on Kubernetes cluster.
+The existing way of executing training through [CML](cml.dev) (CI/CD) works
+  great for handing off ML models between team members but is not the best
+  approach in an experimentation phase since CI/CD creates an overhead on the
+  execution time and it creates some footprint/records in CI/CD that is
+  a bit overwhelming when the number of experiments is huge (10s, 100s or
+  1000s).
+This feature can save time for users and improve ML experimentation experience.
+
+## Scenarios
+
+1. **Run exp.** Allocate a cloud istance (with a specified configuration) and
+  execute an ML experiment `dvc exp run` (or `dvc repro`) on it.
+2. **Run stage.** Allocate a cloud istance and execute a particular stage of
+   an experiment on it.
+3. **Hot instance.** Execute on an existing running instance without instance
+   allocation.
+4. **Remote machine.** Execute in a remote machine (cloud or on-premise one).
+5. **Instance pool.** Execute on one of the existing running instances.
+6. **Exp queue** Execute a set of experiments (from exp queue for example) in
+   an instance pool or a single instance.
```

Comment:
To what extent would DVC utilize pools? Are we thinking of parallel execution/ distributed computing? Or for now focusing on single instances (which may be obtained from a pool)?

Reply:
Single instance or a pool of instances for parallel execution. But it is not about distributed learning. (if I understood the question correctly).